### PR TITLE
Add putFileTo method to HttpTest

### PR DESCRIPTION
**Pull request by jaruzafa**
Hi,
I've added a new putFileTo method to HttpTest, just to test PUT requests with a file. An acceptance test has been included too.

### CODE REVIEW

File: src/main/java/nl/hsac/fitnesse/fixture/slim/HttpTest.java [hunk 1]
```diff
@@ -356,6 +356,40 @@ protected boolean putToImpl(String body, String serviceUrl) {
...
+
+        String filePath = getFilePathFromWikiUrl(fileName);
+        File file = new File(filePath);
+        if (!file.exists()) {
+            throw new StopTestException(false, "File " + filePath + " not found.");
+        }
+
+        try {
+            response.setRequest(fileName);
+            storeLastCall("PUT_FILE", serviceUrl);
```
Comment (fhoeben) [inline src/main/java/nl/hsac/fitnesse/fixture/slim/HttpTest.java]: Please also update `repeatLastCall()` to support PUT_FILE

File: src/main/java/nl/hsac/fitnesse/fixture/slim/HttpTest.java [hunk 2]
```diff
@@ -356,6 +356,40 @@ protected boolean putToImpl(String body, String serviceUrl) {
         return sendToImpl(body, serviceUrl, getContentType(), "PUT");
     }
 
+    /**
+     * Sends a file by HTTP PUT body to service endpoint.
+     * @param fileName fileName to post
+     * @param serviceUrl service endpoint to send body to.
+     * @return true if call could be made and response did not indicate error.
+     */
+    public boolean putFileTo(String fileName, String serviceUrl) {
```
Comment (fhoeben) [inline src/main/java/nl/hsac/fitnesse/fixture/slim/HttpTest.java]: Please also add this method to the documentation in wiki/FitNesseRoot/HsacExamples/SlimTests/HttpTests.wiki

File: src/main/java/nl/hsac/fitnesse/fixture/slim/HttpTest.java [hunk 3]
```diff
@@ -356,6 +356,40 @@ protected boolean putToImpl(String body, String serviceUrl) {
...
+        return putFileToImpl(fileName, serviceUrl);
+    }
+
+
+    protected boolean putFileToImpl(String fileName, String serviceUrl) {
```
Comment (fhoeben) [inline src/main/java/nl/hsac/fitnesse/fixture/slim/HttpTest.java]: Can you see whether you can reduce code duplication with `postFileToImpl()`?

File: src/main/java/nl/hsac/fitnesse/fixture/slim/HttpTest.java (hunk 2 above)
Comment (jaruzafa) [reply src/main/java/nl/hsac/fitnesse/fixture/slim/HttpTest.java]: Done!

File: src/main/java/nl/hsac/fitnesse/fixture/slim/HttpTest.java (hunk 3 above)
Comment (jaruzafa) [reply src/main/java/nl/hsac/fitnesse/fixture/slim/HttpTest.java]: Sure, I've added a sendFileImpl common function and refactored postFileImpl and putFileImpl. No more code duplication

**Comment by jaruzafa** [comment]
I've added PUT_FILE to repeatLastCall too

File: src/main/java/nl/hsac/fitnesse/fixture/slim/HttpTest.java (hunk 1 above)
Comment (jaruzafa) [reply src/main/java/nl/hsac/fitnesse/fixture/slim/HttpTest.java]: Done!

**Pull request merged**